### PR TITLE
Vision: Fixed crash when image is null

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*
#1207 

*Description of changes:*

- Fixed crash when images are null in train or test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by gradientsky — APPROVED** [review]

File: tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py [hunk 1]
```diff
@@ -127,9 +146,44 @@ def _fit(self,
...
+                y_pred_proba = np.zeros(len(X))
+            else:
+                y_pred_proba = np.zeros((len(X), self.num_classes))
+            X = X.reset_index(drop=True)
+            null_indices = X['image'] == ''
+            X = X[~null_indices]
+            null_indices_rows = list(null_indices[null_indices].index)
+            non_null_indices_rows = list(null_indices[~null_indices].index)
+            y_pred_proba[null_indices_rows] = self._dummy_pred_proba
+            y_pred_proba[non_null_indices_rows] = self.model.predict_proba(X, as_pandas=False)
```
Comment (gradientsky) [inline tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py:162]: is it necessary to create lists? I think updates can be done directly with index values using iloc

File: tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py [hunk 2]
```diff
@@ -127,9 +146,44 @@ def _fit(self,
...
+    def _compute_dummy_pred_proba(self, y):
+        num_classes = self.num_classes
+        if self.problem_type in [BINARY, MULTICLASS]:
+            dummies = pd.get_dummies(y)
+            dummy_columns = set(list(dummies.columns))
+            if len(dummies.columns) < num_classes:
+                for c in range(num_classes):
+                    if c not in dummy_columns:
+                        dummies[c] = 0
+            dummies = dummies[list(range(num_classes))]
```
Comment (gradientsky) [inline tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py:178]: dummies = dummies.iloc[range(num_classes)]

File: tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py (hunk 1 above)
Comment (Innixma) [reply tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py:162]: y_pred_proba is not a DataFrame, its a numpy ndarray, it doesn't have an index or iloc.

File: tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py (hunk 2 above)
Comment (Innixma) [reply tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py:178]: That isn't what I am trying to do, I am re-ordering the column names so it ensures they are in the same order as in predict_proba output.

**Comment by szha** [comment]
Job PR-1210-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1210/1/index.html

**Pull request merged**